src/librustc_incremental/persist/directory.rs

```diff
@@ -31,17 +34,66 @@ pub struct DefPathIndex {
 pub struct DefIdDirectory {
     // N.B. don't use Removable here because these def-ids are loaded
     // directly without remapping, so loading them should not fail.
-    paths: Vec<DefPath>
+    paths: Vec<DefPath>,
+
+    // For each crate, saves the crate-name/disambiguator so that
+    // later we can match crate-numbers up again.
+    krates: Vec<KrateInfo>,
+}
+
+#[derive(Debug, RustcEncodable, RustcDecodable)]
+pub struct KrateInfo {
```

Comment:
> Why KrateInfo instead of CrateInfo?

Because `crate` is a keyword... but yeah I realize now we often use `C` when we can (personally I would just always use `K`...).